src/lib.rs

```diff
@@ -823,6 +859,45 @@ impl<T: Eq> PartialEq for Grid<T> {
 
 impl<T: Eq> Eq for Grid<T> {}
 
+
+pub struct GridRowIter<'a, T>{grid: &'a Grid<T>, row_index: usize}
+pub struct GridColIter<'a, T>{grid: &'a Grid<T>, col_index: usize}
+
+impl<'a, T> Iterator for GridRowIter<'a, T> {
+    type Item = Iter<'a, T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let rows = self.grid.rows();
+        let row_index = self.row_index;
+
+        if !(0..rows).contains(&row_index) {
```

Comment:
Love this syntax. Learned something here. Did not know that ranges have a contains function. Very readable